ConsoleApp/AbstractList.cs

```diff
@@ -0,0 +1,21 @@
+namespace ConsoleApp
+{
+    public abstract class AbstractList<T>
```

Comment:
it's not an abstract list since it has the method `Menu`. I don't think you need it, List<T> can be part of `Menu` class